library/common/extensions/filters/http/socket_selection/filter.cc

```diff
@@ -0,0 +1,33 @@
+#include "library/common/extensions/filters/http/socket_selection/filter.h"
+
+#include "envoy/server/filter_config.h"
+
+#include "library/common/network/mobile_utility.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace SocketSelection {
+
+Http::FilterHeadersStatus SocketSelectionFilter::decodeHeaders(Http::RequestHeaderMap&, bool) {
+  ASSERT(decoder_callbacks_);
```

Comment:
nit: not sure this assert adds much value, same below.

Reply:
Agreed, it's mostly a holdover from when we were less certain about the contract on when this will definitively be set. I'll go ahead and remove.